src/main/java/org/candlepin/async/tasks/InactiveConsumerCleanerJob.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.model.ConsumerCurator;
+
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+
+
+
+/**
+ * If enabled, the InactiveConsumerCleanerJob periodically runs and removes inactive consumers
+ * based on their last checked in date and last updated dated. The identity certificate and
+ * content access certificate are removed and their serials are revoked.
+ */
+public class InactiveConsumerCleanerJob implements AsyncJob {
```

Comment:
And actually, you could even throw the entitlement variable there: an old lastcheckin consumer with an entitlement, and the other without entitlement

Reply:
I'm currently working on adding the spec tests

Reply:
I added a spec test however I wasn't able to test the null last checked in date with an outdated last updated time because the update time is automatically set at the current date like you said.